### ISSUE TITLE
Ability to preserve quick search across views

### ISSUE THREAD

**Issue by dstillman** [zotero/zotero#2936]
https://forums.zotero.org/discussion/101403/feature-request-keep-text-in-quick-search and many other threads going back to the beginning, but I can't find another issue for this

Various possibilities: pin like Thunderbird, search history (auto-expanding on focus?), button to switch to library root if no results

**Comment by qqobb** [comment]
Perhaps a checkbox in the drop-down menu of Quick Search?

```
   • Title, Creator, Year
     All Fields & Tags
     Everything
 [x] Search in Library
```

With the "Search in Library" option checked, the grayed-out mode indicator text in the empty Quick Search box could show the library's name:

```
Title, Creator, Year (My Library)
```

(BTW, there's a related Zutilo issue: https://github.com/wshanks/Zutilo/issues/118.)

**Comment by adomasven** [comment]
Just noticed that thunderbird has a nice solution for this

![image](https://github.com/zotero/zotero/assets/5899315/7e7e3a4d-b016-43df-9917-c9b4386303f8)

**Comment by dstillman** [comment]
Yeah, I mention TB above — we've always known about that. The problem is that it would require permanent toolbar space for something that many people won't ever toggle. I think a lot of people would just want to leave this on all the time.

I believe I started implementing a quick-search menu option at some point — on the grounds that it's out of the way if you don't ever use it, but accessible enough that if you generally prefer it off you could turn it on before switching to another collection — but I got hung up on wording that wasn't absurdly long for the menu. "Keep search active when switching collections" would cause the menu to be way too wide. I guess maybe "Pin search term" could work, and people can figure it out?

But my concern with a toggle is that it's just annoyingly fiddly. I also suspect that it's just unnecessary and people almost always just want to repeat the current search in the library root (which I think nearly all of the forum posts have said), in which case we could probably come up with a better way to make that possible.

**Comment by scfrank** [comment]
This is a daily pain-point for me too and I'm glad/hope there are efforts to fix it!

Some thoughts.
1) Personally I would want to keep the filter active across collections etc until I remove the search, not remember to toggle "pin search" every time. 
2) This should be possible to do from the keyboard alone if at all possible (eg Ctrl-F to toggle search on and off).
3) For the context in which you want to repeat the in-collection search in the library root, Thunderbird has a "double enter" mechanism that might work here too.

**Comment by dstillman** [comment]
https://forums.zotero.org/discussion/122469/feature-request-if-search-returns-nothing-offer-searching-the-whole-library

**Comment by abaevbog** [comment]
https://forums.zotero.org/discussion/125916/feature-request-keep-search-box-text-persistent-when-navigating-collections

**Comment by abaevbog** [comment]
I've been seeing a number of fresh requests for this lately (new one: https://forums.zotero.org/discussion/126053/feature-requested-keep-search-filter-when-switching-folder), so wanted to give it a shot as well. For now, I added a new `menuitem` into the dropdown of the quick search. Clicking that item will toggle the pref to retain the quick search value when collections are changed. The phrasing for that `menuitem` is indeed one of the biggest questions, since it needs to be descriptive but not too verbose. I went with "Apply to All Collections", since it's brief but I think gets the message across that if you go to another collection, this search query will still be applied.

How does this look?

https://github.com/user-attachments/assets/dd405780-295d-4b94-8230-e27efc70b03a

**Comment by adomasven** [comment]
I think this is 100% better than not supporting it at all. The wording seems clear enough to not cause excess confusion.
